### DISCUSSION TITLE
Beta 0.4.1

### DISCUSSION THREAD

**Discussion by ourownstory** [ourownstory/neural_prophet#827] [Feature Requests & Feedback]
TBD

<hr /><em>This discussion was created from the release <a href='https://github.com/ourownstory/neural_prophet/releases/tag/0.4.1'>Beta 0.4.1</a>.</em>

**Comment by rbeldagarcia** [comment]
Hello! Thanks for the update, some features are very useful like to allow create a global model with a unique data frame.

I write this comment to report some observation of the new version.  I have tried to migrate my models written in 0.3.2 to 0.4.1 and I have found some issues.
In particular, these models are Global model with events, and the problems appear in: 
Could you confirm if the requirements.txt is updated? I would like to validate that my environment is correct before reporting this as a bug. In particular, I have tried with:

pandas=1.3.5
neuralprophet=0.4.1
numpy=1.21.6

I have tried to create a new environment with requirements.txt but it reports a error trying to install torch>=1.8.0

df_utils.py (line 1006)    

```
    for event in config_events.keys():
        event_feature = pd.Series([0.0] * df.shape[0])
        event_feature.index = df.index  #added to fix the errors, without this it reports NaN when the events are added to the dataframe in global model. 
        				 #The problem is that if df.index != event_feature.index then it return NaN when try to do: df[event] = event_feature


        # events_df may be None in case ID from original df is not provided in events df
        if events_df is None:
            dates = None
        else:
            dates = events_df[events_df.event == event].ds
            event_feature.iloc[df.ds.isin(dates)] = 1.0 #Added 'iloc' to support events that are only present in some 'ID' groups
        df[event] = event_feature
    return df
```
    
I attach you the code and a screen shoot 

[Neuralprophet041.zip](https://github.com/ourownstory/neural_prophet/files/9818669/Neuralprophet041.zip)

**Reply by ourownstory** [reply]
converted to issue #879
